app/src/componentLibrary/dropdown/dropdown.jsx

```diff
@@ -77,30 +134,52 @@ export const Dropdown = ({
     return displayedValue;
   };
 
-  const handleChange = (option) => {
-    if (option.disabled) {
+  const handleToggleButtonKeyDown = (event) => {
+    const { keyCode } = event;
+    if (OPEN_DROPDOWN_KEY_CODES_MAP[keyCode] && !isOpened) {
+      event.preventDefault();
+      setHighlightedIndex(defaultHighlightedIndex);
+      updatePosition.current();
+      setOpened(true);
+      onFocus();
+    }
+  };
+
+  const handleKeyDownMenu = (event) => {
+    const { keyCode } = event;
+    if (keyCode === ENTER_KEY_CODE) {
+      const option = options[highlightedIndex];
+      handleChange(option);
+      setOpened(false);
+      onBlur();
       return;
     }
-    onChange(option.value);
-    setOpened(!isOpened);
+
+    if (CLOSE_DROPDOWN_KEY_CODES_MAP[keyCode]) {
+      event.stopPropagation();
+      setOpened(false);
+      onBlur();
+    }
   };
 
   const renderOptions = () =>
-    options.map((option) => {
-      const isSelected =
-        option.value === ((value && value.value) !== undefined ? value.value : value);
-      return (
-        <DropdownOption
-          key={option.value}
-          selected={isSelected}
-          onChange={option.disabled ? null : () => handleChange(option)}
-          variant={variant}
-          option={option}
-          render={renderOption}
-          isOpened={isOpened}
-        />
-      );
-    });
+    options.map((option, index) => (
+      <DropdownOption
+        key={option.value}
+        index={index}
+        {...getItemProps({
+          item: option,
+          index,
+          selected: option.value === selectedItem.value,
+          variant,
+          option,
+          highlightHovered: highlightedIndex === index,
+          render: renderOption,
+          onChange: option.disabled ? null : () => handleChange(option),
+          onMouseEnter: setHighlightedIndex,
```

Comment:
You can provide `() => setHighlightedIndex(index)` right here, so the index prop no longer required inside of `DropdownOption`.